File: gateway/src/main/java/com/redroundrobin/thirema/gateway/models/Sensor.java
package com.redroundrobin.thirema.gateway.models;

import com.fasterxml.jackson.annotation.JsonIgnore;

import java.security.SecureRandom;

public class Sensor {
  private final int sensorId;
  private long timestamp;
  private int data;
  private final boolean cmdEnabled;

  public Sensor(int sensorId, int data) {
    this.sensorId = sensorId;
    this.timestamp = 0;
    this.data = data;
    this.cmdEnabled = false;
  }

  public Sensor(int sensorId, int data, boolean cmdEnabled) {
    this.sensorId = sensorId;
    this.timestamp = 0;
    this.data = data;
    this.cmdEnabled = cmdEnabled;
  }

  public int getSensorId() {
    return sensorId;
  }

  @JsonIgnore
  public int getData() {
    if (!cmdEnabled) {
      SecureRandom rand = new SecureRandom();
      return rand.nextBoolean() ? data + rand.nextInt(2) : data - rand.nextInt(2);
    } else {
      return data;
    }
  }

  public long getTimestamp() {
    return timestamp;
  }

  public void setTimestamp(long timestamp) {
    this.timestamp = timestamp;
  }

  public void setData(int data) {
    this.data = data;
  }
}

File: device-samples/untitled/lib/models/sensor.cpp
//
// Created by Fritz on 24/04/2020.
//

#include "sensor.h"
sensor::sensor(int sensorId, int arduinoPort) : sensorId(sensorId), arduinoPort(arduinoPort) {}

sensor::sensor(int sensorId, int arduinoPort, bool cmdEnabled, bool toggleable) : sensorId(sensorId),
  arduinoPort(arduinoPort), cmdEnabled(cmdEnabled), toggleable(toggleable) {}

int sensor::getArduinoPort() const {
  return arduinoPort;
}

int sensor::getData() const {
  return data;
}

int sensor::getSensorId() const {
  return sensorId;
}

bool sensor::isCmdEnabled() const {
  return cmdEnabled;
}

bool sensor::isToggleable() const {
  return toggleable;
}

void sensor::setData(int data) {
  sensor::data = data;
}
